core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -333,32 +351,64 @@ object LAFuture {
     }
   }
 
-
   /**
-   * Collect all the future values into the aggregate future
-   * The returned future will be satisfied when all the
-   * collected futures are satisfied
+   * Given handlers for a value's success and failure and a set of futures, runs
+   * the futures simultaneously and invokes either success or failure callbacks
+   * as each future completes. When all futures are complete, if the handlers
+   * have not either satisfied or failed the overall result, `allValuesCompleted`
+   * is called to complete it. If it *still* isn't complete, the overall result
+   * is failed with an error.
+   *
+   * Note that the success and failure functions are guaranteed to be run in a
+   * thread-safe manner. Each is passed the value, the result future, the
+   * accumulating `ArrayBuffer`, and the index of the future that has been
+   * completed. For the failure handler, the value is the `Box` of the failure.
    */
-  def collect[T](future: LAFuture[T]*): LAFuture[List[T]] = {
-    val result = new LAFuture[List[T]]
-    if (future.isEmpty) {
-      result.satisfy(Nil)
+  def collect[T, A](
+    valueSucceeded: (T, LAFuture[A], ArrayBuffer[Box[T]], Int)=>Unit,
+    valueFailed: (Box[Nothing], LAFuture[A], ArrayBuffer[Box[T]], Int)=>Unit,
+    allValuesCompleted: (LAFuture[A], ArrayBuffer[Box[T]])=>Unit,
```

Comment:
Well, we're possibly in some uncanny valley where the type and the intended use are tightly coupled. Unlike a `String` this exact sequence of (arugments, result) only have one possible connotation in this file and enabling a human to identify that sequence correctly across multiple occurrences has value. "Do function X and Y take the same argument at some point?" is a question we frequently have to answer as developers, and function signatures in parameter lists makes answering that more challenging.

That said, I'm not interested in bike shedding until we're both blue in the face. Renaming the parameters to more clearly indicate the event-handling nature would satisfy me enough to rubber stamp this PR. =)